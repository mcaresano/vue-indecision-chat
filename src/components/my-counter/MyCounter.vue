<template>
  <section>
    <hr>
    <h3>Counter: {{ counter }}</h3>
    <h3>Square: {{ counter + squareCounter }}</h3>
    <button class="btn" @click="counter++">+1</button>
    <button class="btn" @click="counter--">-1</button>
  </section> 
</template>


<script lang="ts" setup>
  import { useCounter } from '../../composable/useCounter';
  const {counter, squareCounter} = useCounter()

</script>

<style scoped>
 .btn {
  @apply p-5 bg-blue-500 rounded hover:bg-slate-200 m-5
  }
</style>
