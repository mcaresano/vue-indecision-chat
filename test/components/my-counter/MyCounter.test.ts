import { describe, expect, test } from 'vitest'
import MyCounter from '@/components/my-counter/MyCounter.vue'
import { mount } from '@vue/test-utils'

describe('<MyCounter/>', () => {
  let value = 5
  const wrapper = mount(MyCounter, {
    props: {
      value
    }
  })
  test('should match snapshot', () => {
    // crea el snapshot de los componentes para comparar algÃºn cambio posterior. (Renderizado de componente)
    expect(wrapper.html()).toMatchSnapshot()  
  })

  test('renders the counter value correctly', () => {
    value = 5
    // valida por el contenido de un elemento html. (Renderizado de componente)
    expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`) 
  })

  test('should test label with contain data-testId in the component', () => {
    value = 30
    // valida por el contenido de un elemento html con el agregado de un id personalizado. (Renderizado de componente)
    expect(wrapper.find('[data-test-id="square-label"]').text()).toContain(`Square: ${value}`)
  })

  test('should all h3 labels and values', () => {
    value = 5
    // Valida por el contenido de cada elemento html seleccionado. (Renderizado de componente)
    const [counterLabel, squareLabel] = wrapper.findAll('h3')
    expect(counterLabel.text()).toContain(`Counter: ${value}`)
    expect(squareLabel.text()).toContain(`Square: ${value * 6}`)
  })
})
