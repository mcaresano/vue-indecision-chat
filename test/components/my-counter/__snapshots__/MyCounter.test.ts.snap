// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MyCounter/> > should match snapshot 1`] = `
"<section data-v-4f95d0c9="" value="5">
  <hr data-v-4f95d0c9="">
  <h3 data-v-4f95d0c9="" class="btn">Counter: 5</h3>
  <h3 data-v-4f95d0c9="" data-test-id="square-label" class="btn">Square: 30</h3><button data-v-4f95d0c9="" class="btn">+1</button><button data-v-4f95d0c9="" class="btn">-1</button>
</section>"
`;
